lib/Sema/CSDiag.cpp

```diff
@@ -4699,16 +4699,33 @@ static bool diagnoseImplicitSelfErrors(Expr *fnExpr, Expr *argExpr,
                                    base);
 
     calleeInfo.filterList(argType, argLabels);
-    if (calleeInfo.closeness != CC_ExactMatch)
+
+    auto diagnostic = diag::member_shadows_global_function_near_match;
+    switch (calleeInfo.closeness) {
+    case CC_Unavailable:
+    case CC_Inaccessible:
+    case CC_SelfMismatch:
+    case CC_NonLValueInOut:
+    case CC_GeneralMismatch:
+    case CC_ArgumentCountMismatch:
+    case CC_ArgumentLabelMismatch:
       return false;
 
+    case CC_ExactMatch:
+      diagnostic = diag::member_shadows_global_function;
+      break;
+
+    default:
```

Comment:
Nitpick: I think it makes sense to enumerate all the cases here, rather than assume any new ones will also be fine for this diagnostic.